File: app/src/main/java/com/itachi1706/cheesecakeutilities/Modules/MSLIntegration/model/MSLData.java
package com.itachi1706.cheesecakeutilities.Modules.MSLIntegration.model;

/**
 * Created by Kenneth on 14/1/2019.
 * for com.itachi1706.cheesecakeutilities.Modules.MSLIntegration.model in CheesecakeUtilities
 */
public class MSLData {

    private Subjects[] subjects;
    private Task[] tasks;
    private Exam[] exams;

    public Subjects[] getSubjects() {
        return subjects;
    }

    public Task[] getTasks() {
        return tasks;
    }

    public Exam[] getExams() {
        return exams;
    }

    public class Subjects {
        private String guid, name;

        public String getGuid() {
            return guid;
        }

        public String getName() {
            return name;
        }
    }

    public class Task {
        private String guid, type, title, detail, due_date;
        private double timestamp;
        private String completed_at, subject_guid, exam_guid;
        private String examString;
        private int progress;

        public String getGuid() {
            return guid;
        }

        public String getType() {
            return type;
        }

        public String getTitle() {
            return title;
        }

        public String getDetail() {
            return detail;
        }

        public String getDue_date() {
            return due_date;
        }

        public double getTimestamp() {
            return timestamp;
        }

        public String getCompleted_at() {
            return completed_at;
        }

        public String getSubject_guid() {
            return subject_guid;
        }

        public int getProgress() {
            return progress;
        }

        // If revision

        public void setExamString(String examString) {
            this.examString = examString;
        }

        public String getExam_guid() {
            return exam_guid;
        }

        public String getExamString() {
            return examString;
        }
    }

    public class Exam {
        private String guid, module, date;
        private int duration;
        private boolean resit;
        private String seat, room, subject_guid;

        public String getGuid() {
            return guid;
        }

        public String getModule() {
            return module;
        }

        public String getDate() {
            return date;
        }

        public int getDuration() {
            return duration;
        }

        public boolean isResit() {
            return resit;
        }

        public String getSeat() {
            return seat;
        }

        public String getRoom() {
            return room;
        }

        public String getSubject_guid() {
            return subject_guid;
        }
    }
}

File: app/src/main/java/com/itachi1706/cheesecakeutilities/Modules/MSLIntegration/model/MSLData.kt
package com.itachi1706.cheesecakeutilities.Modules.MSLIntegration.model

import androidx.annotation.Keep

/**
 * Created by Kenneth on 14/1/2019.
 * for com.itachi1706.cheesecakeutilities.Modules.MSLIntegration.model in CheesecakeUtilities
 */
class MSLData {

    val subjects: Array<Subjects>? = null
    val tasks: Array<Task>? = null
    val exams: Array<Exam>? = null

    @Keep
    inner class Subjects {
        val guid: String? = null
        val name: String? = null
    }

    inner class Task {
        val guid: String? = null
        val type: String? = null
        val title: String? = null
        val detail: String? = null
        val due_date: String? = null
        val timestamp: Double = 0.toDouble()
        val completed_at: String? = null
        val subject_guid: String? = null
        val exam_guid: String? = null
        val progress: Int = 0

        // If revision
        var examString: String? = null
    }

    inner class Exam {
        val guid: String? = null
        val module: String? = null
        val date: String? = null
        val duration: Int = 0
        val isResit: Boolean = false
        val seat: String? = null
        val room: String? = null
        val subject_guid: String? = null
    }
}
